sdk/storage/Azure.Storage.DataMovement/src/CommitChunkHandler.cs

```diff
@@ -100,28 +140,61 @@ private async Task CommitBlockEvent(StageChunkEventArgs args)
         {
             if (args.Success)
             {
-                Interlocked.Add(ref _bytesTransferred, args.BytesTransferred);
-                // Use progress tracker to get the amount of bytes transferred
-                if (_bytesTransferred == _expectedLength)
-                {
-                    // Add CommitBlockList task to the channel
-                    await _queueCommitBlockTask().ConfigureAwait(false);
-                }
-                else if (_bytesTransferred > _expectedLength)
-                {
-                    await _updateTransferStatus(StorageTransferStatus.CompletedWithSkippedTransfers).ConfigureAwait(false);
-                    await _invokeFailedEventHandler(
-                            new Exception("Unexpected Error: Amount of bytes transferred exceeds expected length.")).ConfigureAwait(false);
-                }
-                _reportProgressInBytes(_bytesTransferred);
+                // Let's add to the channel, and our notifier will handle the chunks.
+                await _stageChunkChannel.Writer.WriteAsync(args, _cancellationToken).ConfigureAwait(false);
             }
             else
             {
-                // Set status to completed
                 await _invokeFailedEventHandler(new Exception("Failure on Stage Block")).ConfigureAwait(false);
             }
         }
 
+        private async Task NotifyOfPendingStageChunkEvents()
```

Comment:
Will cherry pick these changes out and put in a different PR for future review and testing.

https://github.com/Azure/azure-sdk-for-net/issues/32854